tests/src/Context/ContentTypeContext.php

```diff
@@ -0,0 +1,219 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\nexteuropa\Context\FieldContext.
+ */
+
+namespace Drupal\nexteuropa\Context;
+
+use Behat\Behat\Context\Context;
+use Behat\Gherkin\Node\TableNode;
+
+/**
+ * Context with content type management.
+ */
+class ContentTypeContext implements Context {
+
+  /**
+   * The list of node type before a scenario starts.
+   *
+   * @var array
+   */
+  protected $defaultNodeTypes = array();
+
+  /**
+   * Remember the list of node type.
+   *
+   * @BeforeScenario
+   */
+  public function rememberCurrentNodeTypes() {
+    foreach (node_type_get_types() as $result) {
+      $this->defaultNodeTypes[$result->type] = $result->type;
+    }
+  }
+
+  /**
+   * Removes any node types created after the last list node type remembered.
+   *
+   * @AfterScenario
+   */
+  public function resetNodeTypes() {
+    if (empty($this->defaultNodeTypes)) {
+      return;
+    }
+
+    foreach (node_type_get_types() as $result) {
+      if (!in_array($result->type, $this->defaultNodeTypes)) {
+        node_type_delete($result->type);
+      }
+    }
+    $this->defaultNodeTypes = array();
+  }
+
+  /**
+   * Add a field to a specified content type.
+   *
+   * @param string $arg1
+   *    The machine name of the content type to which attaching the field.
+   * @param TableNode $settings
+   *    Two columns table containing field settings.
+   *
+   * @Given a field with the following settings is added to the :arg1 type:
+   *
+   * @throws \InvalidArgumentException
+   *    If the field exist already in the Drupal system.
+   */
+  public function aFieldWithTheFollowingSettingsIsAddedToTheType($arg1, TableNode $settings) {
```

Comment:
If this methods aim to create fields and field_groups, I believe they should be added into the FieldContext instead.